tests/system/test_agent_keep_alive/test_agent_keep_alive.py

```diff
@@ -0,0 +1,339 @@
+def callback_sending_keep_alive(line):
+    global info
+    match = re.match(r'.*Sending keep alive.*', line)
+    if match:
+        keep_alive = match.group(0)
+        info = keep_alive.split('#!-')[1]
+        return keep_alive
+
+
+def callback_sending_merged(line):
+    global merged
+    match = re.match(r'.*merged.mg*', line)
+    if match:
+        merged = match.group(0)
+        return merged
+
+
+def callback_sending_agent_ip(line):
+    global agent_ip
+    match = re.match(r'.*agent_ip.*', line)
+    if match:
+        agent_ip = match.group(0)
+        return agent_ip
+
+
+def callback_reading_keep_alive(line):
+    global reading
+    match = re.match(r'.*inserting.*', line)
+    if match:
+        reading = match.group(0)
+        reading = reading.split("'")[1]
+        return match.group(0)
+
+
+def callback_reading_merged(line):
+    global reading_merged
+    match = re.match(r'.*merged.mg*', line)
+    if match:
+        reading_merged = match.group(0)
+        return reading_merged
+
+
+def callback_reading_agent_ip(line):
+    global reading_agent_ip
+    match = re.match(r'.*agent_ip.*', line)
+    if match:
+        reading_agent_ip = match.group(0)
+        return reading_agent_ip
+
+
+def callback_inserting_keep_alive(line):
+    global inserting
+    match = re.match(r'.*reading.*', line)
+    if match:
+        inserting = match.group(0)
+        inserting = inserting.split("'")[1]
+        return match.group(0)
+
+
+def callback_inserting_merged(line):
+    global inserting_merged
+    match = re.match(r'.*merged.mg*', line)
+    if match:
+        inserting_merged = match.group(0)
+        return inserting_merged
+
+
+def callback_inserting_agent_ip(line):
+    global inserting_agent_ip
+    match = re.match(r'.*agent_ip.*', line)
+    if match:
+        inserting_agent_ip = match.group(0)
+        return inserting_agent_ip
+
+
+def get_agent_keep_alive():
+    for file in os.listdir(tmp_path):
+        if file == 'wazuh-agent1_ossec.log.tmp':
+            log_monitor = FileMonitor(os.path.join(tmp_path, file))
+            log_monitor.start(timeout=30, callback=callback_sending_keep_alive)
+            log_monitor.start(timeout=30, callback=callback_sending_merged)
+            log_monitor.start(timeout=30, callback=callback_sending_agent_ip)
+
+
+def get_manager_received_keep_alive():
+    for file in os.listdir(tmp_path):
+        if file == 'wazuh-manager_ossec.log.tmp':
+            log_monitor = FileMonitor(os.path.join(tmp_path, file))
+            log_monitor.start(timeout=30, callback=callback_reading_keep_alive)
+            log_monitor.start(timeout=30, callback=callback_reading_merged)
+            log_monitor.start(timeout=30, callback=callback_reading_agent_ip)
+            log_monitor.start(timeout=30, callback=callback_inserting_keep_alive)
+            log_monitor.start(timeout=30, callback=callback_inserting_merged)
+            log_monitor.start(timeout=30, callback=callback_inserting_agent_ip)
```

Comment:
Replace hardcoded timeouts with meaningful variables. If you need to perform some checkings with gathered fields of the regex, return a dictionary instead of using global variables.